src/RelationalBehavior.php

```diff
@@ -113,40 +118,71 @@ public function saveRelations($event)
     {
         /** @var ActiveRecord $model */
         $model = $event->sender;
+        $position = 0;
 
         $relatedRecords = $model->getRelatedRecords();
         foreach ($relatedRecords as $relationName => $relationRecords) {
             $activeQuery = $model->getRelation($relationName);
             if (!empty($activeQuery->via)) { // works only for many-to-many relation
                 list($junctionTable, $primaryModelColumn, $relatedModelColumn) = $this->parseQuery($activeQuery);
-
                 $junctionRows = [];
                 $relationPks = ArrayHelper::getColumn($relationRecords, array_keys($activeQuery->link)[0], false);
                 $passedRecords = count($relationPks);
-                $relationPks = array_filter($relationPks);
+
+                if ($this->sortSettings[$relationName] &&
+                    isset($this->sortSettings[$relationName]['sortColumn']) &&
+                    isset(\Yii::$app->request->getBodyParams()[$relationName])
```

Comment:
Thank you for your notice. Code is updated and ready to merge.

2016-05-20 13:46 GMT+03:00 Andrey <notifications@github.com>:

> In src/RelationalBehavior.php
> <https://github.com/E96/yii2-relational-behavior/pull/1#discussion_r64022298>
> :
>
> >                  $junctionRows = [];
> >                  $relationPks = ArrayHelper::getColumn($relationRecords, array_keys($activeQuery->link)[0], false);
> >                  $passedRecords = count($relationPks);
> > -                $relationPks = array_filter($relationPks);
> > +
> > +                if ($this->sortSettings[$relationName] &&
> > +                    isset($this->sortSettings[$relationName]['sortColumn']) &&
> > +                    isset(\Yii::$app->request->getBodyParams()[$relationName])
>
> Access to post breaks single responsibility principle.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/E96/yii2-relational-behavior/pull/1/files/e5084aef8df528a3e0d1e6ef4c03216519768899#r64022298>
>